[EDI] UDES-11 - Gateway menu is slow when lots of transfers

vvasoyav opened the pull request:
EDI gateway menu is slow down when lots of edi transfers happens. so it is resolved by improve the manage transfer.

Story :: SE-18/11

Signed-off By :: vishal.vasoya@unipart.com

rajeeshknambiar commented on addons/edi/models/edi_gateway.py:
```diff
@@ -193,8 +193,17 @@ def _compute_path_count(self):
+        transfers_data = self.env['edi.transfer'].read_group(
+        [('gateway_id', 'in', self.ids)],
+        ['gateway_id'], ['gateway_id'])
+        transfer_count_dict = {data['gateway_id'][0]: data['gateway_id_count'] for data in transfers_data}
```
Nit: black formatting
```py
        transfers_data = self.env["edi.transfer"].read_group(
            [("gateway_id", "in", self.ids)], ["gateway_id"], ["gateway_id"]
        )
        transfer_count_dict = {
            data["gateway_id"][0]: data["gateway_id_count"] for data in transfers_data
        }
```

vvasoyav replied:
changed.

armandcela replied:
Don't see this changed , maybe i am missing smth

vvasoyav replied:
there are 2 commits can u check please.

armandcela commented on addons/edi/models/edi_gateway.py:
```diff
@@ -193,8 +193,17 @@ def _compute_path_count(self):
+        # We specify the search domain to filter transfers related to the current edi.gateway, 
+        # group the results by gateway_id, and count the number of records for each group.
+    
+        transfers_data = self.env['edi.transfer'].read_group(
```
Nit: We generally define used classes before just for better visibility.
EdiTransfer = self.env["edi.transfer"] 
Then 
trasnfers_data = EdiTransfer.read_group(...

vvasoyav replied:
changed.

armandcela commented on addons/edi/models/edi_gateway.py:
```diff
@@ -193,8 +193,17 @@ def _compute_path_count(self):
     def _compute_transfer_count(self):
         """Compute number of transfers (for UI display)"""
+
+        # We use the read_group method to aggregate data from the edi.transfer model. 
```
Nit: Better to update or add your comment in the doc string, instead of having a docsting and a comment just after docstring

vvasoyav replied:
changed

vvasoyav committed:
[EDI] UDES-11 - Gateway menu is slow when lots of transfers

EDI gateway menu is slow down when lots of edi transfers happens.
so it is resolved by improve the manage transfer.

Story :: SE-18/11

Signed-off-by: vishal vasoya <vishal.vasoya@unipart.com>
